src/traversal.js

```diff
@@ -33,6 +34,7 @@ var dummy;
 var enterNode = function(node) {
   var data = getData(node);
   enterTag(data.nodeName);
+  cleanKeyMap(node);
```

Comment:
This needs to be done on exit, not enter, or we'll still have a memory leak until the next `patch`.